homeassistant/components/hive/climate.py

```diff
@@ -43,21 +45,11 @@ def __init__(self, hivesession, hivedevice):
         """Initialize the Climate device."""
         self.node_id = hivedevice["Hive_NodeID"]
         self.node_name = hivedevice["Hive_NodeName"]
-        self.device_type = hivedevice["HA_DeviceType"]
-        if self.device_type == "Heating":
-            self.thermostat_node_id = hivedevice["Thermostat_NodeID"]
+        self.thermostat_node_id = hivedevice["Thermostat_NodeID"]
         self.session = hivesession
         self.attributes = {}
-        self.data_updatesource = '{}.{}'.format(
-            self.device_type, self.node_id)
-        self._unique_id = '{}-{}'.format(self.node_id, self.device_type)
-
-        if self.device_type == "Heating":
-            self.modes = [STATE_AUTO, STATE_HEAT, STATE_OFF]
-        elif self.device_type == "HotWater":
-            self.modes = [STATE_AUTO, STATE_ON, STATE_OFF]
-
-        self.session.entities.append(self)
+        self.data_updatesource = 'Heating.{}'.format(self.node_id)
+        self._unique_id = '{}-Heating'.format(self.node_id)
```

Comment:
We need to keep the original format so that it aligns to the other Hive components for when new data is updated from the cloud to the cache from one Hive component, all the other Hive components refresh their data from the latest cached data

Reply:
sure, but inside `__init__` is a bad design pattern, so I move it out.